hazelcast/src/main/java/com/hazelcast/internal/config/MemberDomConfigProcessor.java

```diff
@@ -1332,18 +1332,25 @@ private void handleAliasedDiscoveryStrategy(WanBatchPublisherConfig publisherCon
     private void updateConfig(AliasedDiscoveryConfig config, Node node) {
         NamedNodeMap attributes = node.getAttributes();
         for (int a = 0; a < attributes.getLength(); a++) {
-            Node att = attributes.item(a);
+            Node att = attributes.item(a);            
             String value = getTextContent(att).trim();
             if ("enabled".equals(lowerCaseInternal(att.getNodeName()))) {
                 config.setEnabled(getBooleanValue(value));
             } else if (att.getNodeName().equals("connection-timeout-seconds")) {
                 config.setProperty("connection-timeout-seconds", value);
             }
         }
+        LowercaseHandler lowerCaseHandler = new LowercaseHandler();
         for (Node n : childElements(node)) {
+        	lowerCaseHandler.put(n.getLocalName(), "eureka".equals(n.getParentNode().getLocalName()));
             String key = cleanNodeName(n);
```

Comment:
Can't we simply pass the `tag` to this metrod from the caller and do the following without the need for the handler?
```
String key = cleanNodeName(n, !tag.equals("eureka"));

[...]

public static String cleanNodeName(final Node node, final boolean shouldLowerCase) {
[...]
}
```
This seems we put a "local" boolean to a hashmap that we lookup for a few lines later in the same cycle, so the handler and the map feel unnecessary here.

Reply:
cleanNodeName is being used in a lot of places I wanted to have less impact in the codebase

Reply:
I see your point @omidp. You can keep the change local with overriding the `cleanNodeName` method like
```
    public static String cleanNodeName(final Node node) {
        return cleanNodeName(node, true);
    }

    public static String cleanNodeName(final Node node, final boolean shouldLowerCase) {
        final String nodeName = node.getLocalName();
        if (nodeName == null) {
            throw new HazelcastException("Local node name is null for " + node);
        }
        return shouldLowerCase ? StringUtil.lowerCaseInternal(nodeName) : nodeName;
    }
```
So no other code needs to be modified and we follow the common logic for eureka config processing as well. I'd prefer this over introducing a helper class and a map.

Reply:
Ok done